frontend/src/Editor/Components/PDF.jsx

```diff
@@ -58,35 +84,75 @@ export const PDF = React.memo(({ styles, properties, width, height }) => {
           inputRef={(el) => (pageRef.current[index] = el)}
         />
       ))}
-      {pageControls && (
-        <>
-          <div className="pdf-page-controls">
-            <button disabled={pageNumber <= 1} onClick={() => updatePage(-1)} type="button" aria-label="Previous page">
-              ‹
-            </button>
-            <span>
-              {pageNumber} of {numPages}
-            </span>
-            <button
-              disabled={pageNumber >= numPages}
-              onClick={() => updatePage(1)}
-              type="button"
-              aria-label="Next page"
-            >
-              ›
-            </button>
-          </div>
-        </>
-      )}
     </Document>
   );
 
+  async function downloadFile(url, pdfName) {
+    const pdf = await fetch(url);
+    const pdfBlog = await pdf.blob();
+    const pdfURL = URL.createObjectURL(pdfBlog);
+    const anchor = document.createElement('a');
+    anchor.href = pdfURL;
+    anchor.download = pdfName;
+    document.body.appendChild(anchor);
+    anchor.click();
+    document.body.removeChild(anchor);
+    URL.revokeObjectURL(pdfURL);
+  }
   return (
     <div style={{ display: visibility ? 'flex' : 'none', width: width - 3, height }}>
-      <div className="d-flex position-relative h-100" style={{ margin: '0 auto', overflow: 'hidden' }}>
+      <div className="d-flex position-relative h-100 flex-column" style={{ margin: '0 auto', overflow: 'hidden' }}>
         <div className="scrollable h-100 col position-relative" id="pdf-wrapper">
           {url === '' ? 'No PDF file specified' : renderPDF()}
         </div>
+        {!error && !pageLoading && (showDownloadOption || pageControls) && (
+          <div
+            className={`d-flex ${
+              pageControls ? 'justify-content-between' : 'justify-content-end'
+            } py-3 px-3 align-items-baseline bg-white border-top border-light`}
+          >
+            {pageControls && (
+              <>
+                <div className="pdf-page-controls">
+                  <button
+                    disabled={pageNumber <= 1}
+                    onClick={() => updatePage(-1)}
+                    type="button"
+                    aria-label="Previous page"
+                  >
+                    ‹
+                  </button>
+                  <span>
+                    {pageNumber} of {numPages}
+                  </span>
+                  <button
+                    disabled={pageNumber >= numPages}
+                    onClick={() => updatePage(1)}
+                    type="button"
+                    aria-label="Next page"
+                  >
+                    ›
+                  </button>
+                </div>
+              </>
+            )}
+            {showDownloadOption && (
+              <div
+                className="download-icon-outer-wrapper text-dark"
+                style={downloadIconOuterWrapperStyles}
+                onClick={() => downloadFile(url, pdfName)}
```

Comment:
Would it be possible to use `<PDFDownloadLink />` here?
ref: https://react-pdf.org/advanced#on-the-fly-rendering

Reply:
@gondar00 We need to install a `react-pdf/renderer` package to make use of it. So, we avoided that

Reply:
On a closer look @ [`<PDFDownloadLink />`](https://github.com/diegomura/react-pdf/blob/d011983204cf45876594fa361f24b47e86c612c9/packages/renderer/src/dom/PDFDownloadLink.js) it seems like a wrapper around a anchor tag only, can ignore it